core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1040,6 +1040,11 @@ public void listStatus(AlluxioURI path, ListStatusContext context,
           }
           ensureFullPathAndUpdateCache(inodePath);
 
+          if (context.getOptions().getLoadMetadataOnly()) {
```

Comment:
@gpang  is this the correct location to continue if we don't need results returned?

Reply:
@gpang  ping

Reply:
I think this is fine, but I think I would suggest a slightly different approach. The part after this code gathers all the results, and updates metrics. If we return here, we will not get some of the metrics accounting. Instead, it could look something like this:

```java
ensureFullPathAndUpdateCache(inodePath);

auditContext.setSrcInode(inodePath.getInode());
if (!context.getOptions().getLoadMetadataOnly()) {  // this line is new
  DescendantType descendantTypeForListStatus =
    (context.getOptions().getRecursive()) ? DescendantType.ALL : DescendantType.ONE;
  listStatusInternal(context, rpcContext, inodePath, auditContext,
    descendantTypeForListStatus, resultStream, 0);
}    // this line is new
auditContext.setSucceeded(true);
Metrics.FILE_INFOS_GOT.inc();
if (!ufsAccessed) {
  MountTable.Resolution resolution = mMountTable.resolve(inodePath.getUri());
  Metrics.getUfsCounter(mMountTable.getMountInfo(resolution.getMountId())
      .getUfsUri().toString(),
    Metrics.UFSOps.LIST_STATUS).inc();
}
```

Basically, what this is doing is the method is unchanged, but the option only guards the call to `listStatusInternal()` which is the method that fills in all the results. If we don't want results, we only skip calling that method.

What do you think?

Reply:
SGTM, @JySongWithZhangCe  Can you address @gpang 's comments?

Reply:
SGTM, Thx @gpang and @apc999 !